services/frontend/src/scenes/Profile/ProfileLayout.js

```diff
@@ -122,18 +124,85 @@ const PersonalDetails = ({profile}) => {
     );
 };
 
-const Skills = ({sectionName, profile}) => (
-    <>
-        <h2>{sectionName}</h2>
-        <Paper className="profile-page-card profile-card-skills">
-            <SkillBadges
-                displayCount={profile.skills.length}
-                skills={profile.skills}
-            />
-        </Paper>
-    </>
-);
+const Skills = ({sectionName, profile, skills}) => {
+    const [editSkillsDialogOpen, setEditSkillsDialogOpen] = React.useState(false);
+    const [profileUpdated, updateProfile] = useState(profile);
+
+    const openDialog = () => {
+        setEditSkillsDialogOpen(true);
+    };
+
+    const closeDialog = () => {
+        setEditSkillsDialogOpen(false);
+    };
+
+    const handleSubmit = (updatedSkills) => {
+        removeSkills(updatedSkills);
+        addSkills(updatedSkills);
+        closeDialog();
+    };
+
+    const handleCancel = () => {
+        closeDialog();
+    };
+
+    const addSkills = (updatedSkills) => {
```

Comment:
This is a bit wacky, not the most readable, maps skill names to lowercase way more times than required and maps over arrays but doesn't return anything (grounds for using a for loop instead).

Here's an example implementation since github doesn't support multiline reviews...

```js
const addSkills = (updatedSkills) => {
    const tempProfile = profileUpdated;
    const oldSkillsLC = profileUpdated.skills.map((skill) => skill.name.toLowerCase());
    const dataBaseSkillsLC = Object.values(skills).map((skill) => skill.name.toLowerCase());

    /* returns a list of the new skills entered in the dialog */
    const newSkills = updatedSkills.reduce((acc, skillName) => {
        const newSkillNameLC = skillName.toLowerCase();
        if (!oldSkillsLC.includes(newSkillNameLC)) {
            return [...acc, newSkillNameLC];
        }
        return acc;
    }, []);

    /* adds skills from the existing version in the database or creates a new one */
    for (const skillName of newSkills) {
        if (dataBaseSkillsLC.includes(skillName)) {
            /* Adds from Existing Skills List*/
            for (const skill of Object.values(skills)) {
                if (skill.name.toLowerCase() === skillName) {
                    tempProfile.skills.push(skills[skill.id]);
                }
            }
        } else {
            /*Adds new Skill Object */
            const newSkill = new Skill({name: skillName});
            tempProfile.skills.push(newSkill);
        }
    }
    updateProfile(tempProfile);
};
```